src/messaging/ExchangeContext.h

```diff
@@ -165,9 +163,9 @@ class DLL_EXPORT ExchangeContext : public ReferenceCounted<ExchangeContext, Exch
 
     uint16_t GetExchangeId() const { return mExchangeId; }
 
-    void SetChallenge(const uint8_t * value) { memcpy(&mChallenge[0], value, kMsgCounterChallengeSize); }
+    void SetAppState(void * state) { mAppState = state; }
```

Comment:
`Challenge` will be moved to `PeerConnectionState`, please leave it along here, I'll remove it from the message counter PR.

We'd better not introduce a `void*` here.

Reply:
Moving Challenge to PeerConnectionState seems will increase RAM not decrease RAM usage, right?

Reply:
Change `mChallenge` to `mAppState` doesn't decrease RAM usage.

Reply:
Why not? it can save 32 bytes on 32 -bit machine with CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS = 8 and 64 bytes with default value.